pydeseq2/dds.py

```diff
@@ -461,16 +453,11 @@ def fit_MAP_dispersions(self):
         """
 
         # Check that the dispersion prior variance is available. If not, compute it.
-        if not hasattr(self, "prior_disp_var"):
+        if "prior_disp_var" not in self.uns:
             self.fit_dispersion_prior()
 
-        # Exclude genes with all zeroes
-        counts_nonzero = self.counts.loc[:, self.non_zero_genes].values
-        num_genes = counts_nonzero.shape[1]
-
-        X = self.design_matrix.values
-        mu_hat = self._mu_hat.values
-        fit_disps = self.fitted_dispersions.loc[self.non_zero_genes].values
+        # Convert design matrix to numpy for speed
+        X = self.obsm["design_matrix"].values
```

Comment:
same comment as above, consider calling it `design_matrix_np`?

Reply:
done in [2bcc4dc](https://github.com/owkin/PyDESeq2/pull/66/commits/2bcc4dc7abc6d41951f4dcc70263b78a2b44ffdc)